inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,369 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <array>
+#include <fstream>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "ngraph/ops.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "pugixml.hpp"
+#include "transformations/serialize.hpp"
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::Serialize, "Serialize", 0);
+
+namespace {  // helpers
+template <typename T, typename A>
+std::string joinVec(std::vector<T, A> const& vec,
```

Comment:
Ok. I will refactor this at first ocasion. Don't want to retrigger CI right now cause it is very painful this days to have it complete without errors. This PR is close to be merged so I'll do this if there will be some need for next commit, if not then I'll do this in next related PR (where I'm adding experimental operations support to serialization).

Btw. I actually copy-pasted this code from [here](https://github.com/openvinotoolkit/openvino/blob/95cc54c5752b78471125fcdb8fcf11b9284d078d/inference-engine/src/plugin_api/debug.h#L118). Couldn't use it because transformation library doesn't link with ie library. The issue I see is that everyone keeps reimplementing the same thing again and again ... we should think about some static utility library with common code which would be available for all of the code. 

FYI @ilyachur